dymos/load_case.py

```diff
@@ -51,83 +78,101 @@ def load_case(problem, previous_solution):
+            prev_state_path = [s for s in prev_vars if s.endswith(f'{phase_name}.timeseries.states:{state_name}')][0]
+            prev_state_val = prev_vars[prev_state_path]['value']
+            prev_state_units = prev_vars[prev_state_path]['units']
             problem.set_val(state_path,
-                            phase.interpolate(xs=prev_time, ys=prev_state_val,
+                            phase.interpolate(xs=prev_time_val, ys=prev_state_val,
                                               nodes='state_input', kind='slinear'),
                             units=prev_state_units)
 
         # Interpolate the timeseries control outputs from the previous solution onto the new grid.
         for control_name, options in phase.control_options.items():
-            control_path = [s for s in phase_outputs if s.endswith(f'{phase_name}.controls:{control_name}')][0]
-            prev_control_path = [s for s in prev_outputs
+            control_path = [s for s in phase_vars if s.endswith(f'{phase_name}.controls:{control_name}')][0]
+            prev_control_path = [s for s in prev_vars
                                  if s.endswith(f'{phase_name}.timeseries.controls:{control_name}')][0]
-            prev_control_val = prev_outputs[prev_control_path]['value']
-            prev_control_units = prev_outputs[prev_control_path]['units']
+            prev_control_val = prev_vars[prev_control_path]['value']
+            prev_control_units = prev_vars[prev_control_path]['units']
             problem.set_val(control_path,
-                            phase.interpolate(xs=prev_time, ys=prev_control_val,
+                            phase.interpolate(xs=prev_time_val, ys=prev_control_val,
                                               nodes='control_input', kind='slinear'),
                             units=prev_control_units)
 
         # Set the output polynomial control outputs from the previous solution as the value
-        for polynomial_control_name, options in phase.polynomial_control_options.items():
-            polynomial_control_path = [s for s in phase_outputs if
-                                       s.endswith(f'{phase_name}.polynomial_controls:{polynomial_control_name}')][0]
-            prev_polynomial_control_path = [s for s in prev_outputs if
-                                            s.endswith(f'{phase_name}.'
-                                                       f'polynomial_controls:{polynomial_control_name}')][0]
-            prev_polynomial_control_val = prev_outputs[prev_polynomial_control_path]['value']
-            prev_polynomial_control_units = prev_outputs[prev_polynomial_control_path]['units']
+        for pc_name, options in phase.polynomial_control_options.items():
+            polynomial_control_path = [s for s in phase_vars if
+                                       s.endswith(f'{phase_name}.polynomial_controls:{pc_name}')][0]
+            prev_polynomial_control_path = [s for s in prev_vars if
```

Comment:
I think `polynomial_control_path` and `prev_polynomial_control_path` are also the same, so you only need one of them.

Reply:
Could also follow your lead with pc_name, and convert "polynomial_control" to "pc" in these other var names for readability.

Reply:
Again I think we should allow for the case where the path to the phase isn't identical (although that may not be a great assumption that we want to revisit)